src/library/searchquery.cpp

```diff
@@ -166,6 +172,46 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+CrateFilterNode::CrateFilterNode(const QSqlDatabase& database,
+                                 const QString& argument)
+    : m_database(database),
+      m_argument(argument) {
+    getTrackIds();
+}
+
+void CrateFilterNode::getTrackIds() {
+    FieldEscaper escaper(m_database);
+    QString escapedArgument = escaper.escapeString(kSqlLikeMatchAll + m_argument + kSqlLikeMatchAll);
+
+    QSqlQuery query(m_database);
```

Comment:
This can be accomplished much more efficiently:
- New member function: CrateTrackSelectResult CrateStorage::selectCrateTracksSortedByCrateNameLike(const QString& crateName) const
-- Very similar to CrateStorage selectCrateTracksSorted()
-- Instead of providing a single crateId provide a _crateNameLike_ argument and do a sub-select (that we also need for CrateFilterNode::toSql())
- CrateFilterNode: Initialization
-- Perform DB query
-- Ignore crateId()s
-- Collect all distinct trackIds()s in a std::vector
-- Skip duplicate trackId()s resulting from multiple crates matching the LIKE query
- CrateFilterNode::match()
-- Use std::binary_search() on the sorted vector for fast O(log(n)) lookup

Please refer to CrateFeature for a similar use case.

Optimization: Use lazy evaluation, i.e. the first call to match() should trigger the database query and populate the result set once.